### PR TITLE
dnsdist: Split the Lua files, remove deprecated functions

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Split the existing Lua files (dnsdist-lua.cc, dnsdist-lua2.cc) into:
- dnsdist-lua.cc: main one, holding configurations directives
- dnsdist-lua-actions.cc: actions-related code
- dnsdist-lua-bindings.cc: bindings for most structures like ComboAddress, Netmask, etc
- dnsdist-lua-bindings-dnsquestion.cc: bindings for DNSQuestion, DNSResponse
- dnsdist-lua-inspection.cc: inspection-related code (topXX, grepq(), ..)
- dnsdist-lua-rules.cc rules-related code
- dnsdist-lua-vars.cc: constants like DNSClass, DNSTypes, rcodes..

The main goal was to improve readability, but this might also make merges less painful and reduce the memory needed to compile dnsdist.
Note that it does remove functions deprecated in 1.2.0 because they prevented splitting actions from rules.
Comments welcome!

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
I just asked for review but please only review the general idea and how functions and classes are split into files, not actual the code which is just moved around :-)

**Comment by rgacogne** [comment]
rebased to fix conflicts.

File: pdns/dnsdist-lua-rules.cc [hunk 1]
```diff
@@ -0,0 +1,1206 @@
+/*
```
Comment (pieterlexis) [inline pdns/dnsdist-lua-rules.cc]: Although we call this rules, it are selectors... maybe we should rename this in the future :)

File: pdns/dnsdist-lua.cc [hunk 2]
```diff
@@ -786,1098 +524,867 @@ vector<std::function<void(void)>> setupLua(bool client, const std::string& confi
...
-	      rule, 
-		std::make_shared<QPSPoolAction>(limit, pool)  });
-	});
-    });
+  g_lua.writeFunction("setECSOverride", [](bool override) { g_ECSOverride=override; });
 
-  g_lua.writeFunction("setDNSSECPool", [](const std::string& pool) {
-      setLuaSideEffect();
-      warnlog("setDNSSECPool() is deprecated and will be removed in 1.3.0, please use addAction(DNSSECRule(), PoolAction(\"%s\")) instead", pool);
+  g_lua.writeFunction("showDynBlocks", []() {
```
Comment (pieterlexis) [inline pdns/dnsdist-lua.cc]: Maybe this should go to dnsdist-lua-inspection.cc?

File: pdns/dnsdist-lua.cc [hunk 3]
```diff
@@ -786,1098 +524,867 @@ vector<std::function<void(void)>> setupLua(bool client, const std::string& confi
...
-	  rulactions.push_back({rule, 
-		std::make_shared<QPSAction>(lim)});
-	});
+      nmts_t nmg;
+      g_dynblockNMG.setState(nmg);
+      SuffixMatchTree<DynBlock> smt;
+      g_dynblockSMT.setState(smt);
+    });
+
+  g_lua.writeFunction("addDynBlocks",
```
Comment (pieterlexis) [inline pdns/dnsdist-lua.cc]: maybe we need a dnsdist-lua-dynblocks.cc?

**Comment by zeha** [comment]
Is dnsdist-lua.hh also split or does touching it mean recompiling all of those files?

**Comment by rgacogne** [comment]
No, it's not split but there is very few things in there so I don't expect it to be updated often.

**Comment by zeha** [comment]
> No, it's not split but there is very few things in there so I don't expect it to be updated often.

Oh indeed, I thought all Actions have their prototype in there. Sorry for the noise.

**Pull request merged**